src/Handlers/AbstractPaymentMethodHandler.php

```diff
@@ -741,8 +741,8 @@ private function displaySafeErrorMessages(AdyenException $exception)
      */
     private function getSplitStreetAddressHouseNumber(string $address): array
     {
-        $streetFirstRegex = '/(?<streetName>[\w\W]+)\s+(?<houseNumber>\d{1,10}((\s)?\w{1,3})?)$/m';
-        $numberFirstRegex = '/^(?<houseNumber>\d{1,10}((\s)?\w{1,3})?)\s+(?<streetName>[\w\W]+)/m';
+        $streetFirstRegex = '/(?<streetName>[\w\W]+)\s+(?<houseNumber>(\d|\-)+((\s)?\w{1,3})?)$/m';
+        $numberFirstRegex = '/^(?<houseNumber>(\d|\-)+((\s)?\w{1,3})?)\s+(?<streetName>[\w\W]+)/m';
```

Comment:
You can also keep the previous 10 digits constraint and avoid an extra capturing group by using `[\d-]{1,10}`: 
```suggestion
        $streetFirstRegex = '/(?<streetName>[\w\W]+)\s+(?<houseNumber>[\d-]{1,10}((\s)?\w{1,3})?)$/m';
        $numberFirstRegex = '/^(?<houseNumber>[\d-]{1,10}((\s)?\w{1,3})?)\s+(?<streetName>[\w\W]+)/m';
```